src/test/app/Discrepancy_test.cpp

```diff
@@ -0,0 +1,4089 @@
+                "601692A6C6416DBA294F0DA684675A824B201123000000000000000000000000"
+                "04A50590000000000E5C92828261DBAAC933B6309C6F5C72AF020AFD401E5C92"
+                "828261DBAAC933B6309C6F5C72AF020AFD4FF100000000000000000000000000"
+                "000000000000000300000000000000000000000004A50590000000000E5C9282"
+                "8261DBAAC933B6309C6F5C72AF020AFD401E5C92828261DBAAC933B6309C6F5C"
+                "72AF020AFD4FF01A034782E2DBAC4FB82B601CD50421E8EF24F3A00100000000"
+                "000000000000000000000000000000000300000000000000000000000004A505"
+                "90000000000E5C92828261DBAAC933B6309C6F5C72AF020AFD401E5C92828261"
+                "DBAAC933B6309C6F5C72AF020AFD400";
+            auto jrr = env.rpc ("json", "submit", to_string(jrq))[jss::result];
+            auto hash = jrr[jss::tx_json][jss::hash];
+            env.close();
+            Json::Value jrq2;
+            jrq2[jss::binary] = false;
+            jrq2[jss::transaction] = hash;
+            jrq2[jss::id] = 3;
+            jrr = env.rpc ("json", "tx", to_string(jrq2))[jss::result];
+            uint64_t fee = jrr[jss::Fee].asUInt();
+            auto meta = jrr[jss::meta];
+            auto sumPrev = UINT64_C(0);
+            auto sumFinal = UINT64_C(0);
+            for(auto const& an : meta[sfAffectedNodes.fieldName])
+            {
+                Json::Value node;
+                if(an.isMember(sfCreatedNode.fieldName))
+                    node = an[sfCreatedNode.fieldName];
+                else if(an.isMember(sfModifiedNode.fieldName))
+                    node = an[sfModifiedNode.fieldName];
+                else if(an.isMember(sfDeletedNode.fieldName))
+                    node = an[sfDeletedNode.fieldName];
+
+                if(node && node[sfLedgerEntryType.fieldName] == "AccountRoot")
+                {
+                    Json::Value prevFields =
+                        node.isMember(sfPreviousFields.fieldName) ?
+                        node[sfPreviousFields.fieldName] :
+                        node[sfNewFields.fieldName];
+                    Json::Value finalFields =
+                        node.isMember(sfFinalFields.fieldName) ?
+                        node[sfFinalFields.fieldName] :
+                        node[sfNewFields.fieldName];
+                    if(prevFields)
+                        sumPrev += beast::lexicalCastThrow<std::uint64_t>(
+                            prevFields[sfBalance.fieldName].asString());
+                    if(finalFields)
+                        sumFinal += beast::lexicalCastThrow<std::uint64_t>(
+                            finalFields[sfBalance.fieldName].asString());
+                }
+            }
+            // the difference in balances (final and prev) should be the
+            // fee charged
+            BEAST_EXPECT(sumPrev-sumFinal == fee);
+        }
+
+        boost::filesystem::remove_all(path, ec);
```

Comment:
I'll mention this in passing, but I don't necessarily think it's a good idea.  I noticed calls to `lexicalCastThrow`.  If one of those throws then the call to `boost::filesystem::remove_all()` won't happen and you'll leave temporary files behind.  This is both unlikely and not the end of the world.

However, if you wanted to be more confident that the files were always cleaned up, you could use RAII by declaring this just after `path` is constructed (toward the top of the function):
```
        // Make sure the temporaries are cleared away.
        auto removeAll = [&path, &ec] (void*)
            {
                boost::filesystem::remove_all(path, ec);
            };
        std::unique_ptr<void, decltype (removeAll)> guard (&removeAll, removeAll);
```
This is probably way over engineered for a unit test.  I don't expect you to actually do it.